horace_core/utilities/range_add_border.m

```diff
@@ -34,3 +39,21 @@
     abs_tol = abs(tol)*([-ones(1,ndim);ones(1,ndim)]);
     range(:,no_range)  = range(:,no_range)+abs_tol(:,no_range);   
 end
+
+function range = add_eps_border(range_in)
+% Add epsilon-sized border to cut limits to avoud round-off errors
+sig_range = sign(range_in);
+min_border = 1-4*eps*sig_range(1,:);
```

Comment:
eps by definition is the minimal value, so 1 ~=1+/- eps. This is usually system dependent and will be different if you use 32 bit architecture or different C++ compiler options. It currently the same on all our build systems but does not have to (and was not) Where 4 comes is a bit more subtle -- rather from experience to guard against the things like sqrt(1-eps) == 1. May be can be removed if proper tests are written. 

Here I have taken all places where the border was added, collected all code together and put it into single function. 
To be fair, this function should be moved to Herbert, simplified and properly unit tested. I have not done that not to bloat the ticket changes even more. Such changes worth a separate ticket (new unit tests + the changes to existing tests will happen)

Reply:
so the tol border is smaller than any user-specified tol (abs or rel) sized border and is the smallest area guaranteed to contain all the points?

It's not clear whether the calculation of the eps border would be the same as a tol border calculated with a tol=4*eps. The calculation is laid out very differently.

If it is the same, the "calculate border" calculations could be unified when you move it to Herbert. If it's not a comment documenting how it is different (in the n-dimensional space) would be helpful for clarity.

Reply:
`so the tol border is smaller than any user-specified tol (abs or rel) sized border and is the smallest area guaranteed to contain all the points?`
-- yes, I believe that

it indeed looks like if the toll=4*eps it is different from 4*eps border. I have noticed this.  Within this ticket I have collected all border calculations within the code in one place and exposed its oddity. I have not changed the code itself as it will break existing unit tests (against references) and I want to rely on them for my current work. 

The code should be unified but good thinking is necessary to ensure the differences are just an error or have a good reason behind them.

How to clarify this -- I do not know. Probably TODO in the function header is sufficient for now